### PR TITLE
accounts for `pyuvdata.UVCal.cal_style` parameter in `cal_formats`

### PR DESCRIPTION
**Pull request by nkern**
adapted cal_formats to account for new UVCal cal_style parameter
also updated abscal_funcs.gains2calfits to account for this
and added tests to cal_formats

### CODE REVIEW

File: hera_cal/cal_formats.py [hunk 1]
```diff
@@ -112,10 +110,31 @@ def __init__(self, meta, gains, flags=None, DELAY=False, ex_ants=[], appendhist=
         self.time_array = time
         self.integration_time = meta['inttime']
         self.gain_convention = 'divide'
+        self.set_redundant()
```
Comment (plaplant) [inline hera_cal/cal_formats.py]: I am fine with having this be the default value, but we should mention in the documentation that this is the case.

Comment (nkern) [reply hera_cal/cal_formats.py]: ok, added it in

**Comment by nkern** [comment]
adresses #153

**Comment by nkern** [comment]
also addresses #144

**Comment by nkern** [comment]
@plaplant whenever you ok w/ this feel free to merge it in

**Pull request merged**